hazelcast/src/main/java/com/hazelcast/map/impl/proxy/MapProxyImpl.java

```diff
@@ -652,7 +669,16 @@ public Set entrySet(Predicate predicate) {
         if (predicate instanceof PagingPredicate) {
             return queryEngine.queryAllPartitionsWithPagingPredicate(name, (PagingPredicate) predicate, IterationType.VALUE);
         } else {
-            QueryResult result = queryEngine.invokeQueryAllPartitions(name, predicate, IterationType.VALUE);
+            QueryResult result;
+            if (predicate instanceof PartitionPredicate) {
+                PartitionPredicate partitionPredicate = (PartitionPredicate) predicate;
+                int partitionId = getNodeEngine().getPartitionService().getPartitionId(partitionPredicate.getPartitionKey());
+                result = queryEngine.invokeQuerySinglePartition(
+                        name, partitionPredicate.getTarget(), IterationType.VALUE, partitionId);
+            } else {
+                result = queryEngine.invokeQueryAllPartitions(name, predicate, IterationType.VALUE);
+            }
+
             return new QueryResultCollection<V>(
```

Comment:
OK, thx. Let's not move it now, since I'm in the middle of the refactoring of the queryEngine.